### PR TITLE
Refactor shared photos views

### PR DESCRIPTION
**Pull request by jmalm**
Use react-pig for photos in SharedToMe and SharedFromMe.

This required quite a bit of refactoring, but I intentionally left shared album views as untouched as possible. They will need to be updated separately.

Must be merged together with https://github.com/LibrePhotos/librephotos/pull/378

### CODE REVIEW

**Comment by derneuere** [comment]
I get an error when I compile the frontend:
```
librephotos-docker-frontend-1  | ./src/layouts/dataviz/FaceDashboard.js
librephotos-docker-frontend-1  |   Line 725:  'topMenuHeight' is not defined  no-undef
```

File: src/components/CustomSearch.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ import { serverAddress } from "../api_client/apiClient";
 import { SecuredImageJWT } from "./SecuredImage";
 
 var ENTER_KEY = 13;
-var topMenuHeight = 50; // don't change this
+var topMenuHeight = 50; // don't change this. TODO: Should this use TOP_MENU_HEIGHT (=45)?
```
Comment (derneuere) [inline src/components/CustomSearch.js]: I think it should use TOP_MENU_HEIGHT. It does not look broken if I use it in my instance.

**Comment by jmalm** [comment]
Oops, ok. It should now use a constant instead, TOP_MENU_HEIGHT.

**Comment by derneuere** [comment]
Looks now good to me 👍 Thanks again for all the work you did.

**Comment by jmalm** [comment]
Hey, thanks for all _you_ do!

I have some further deduplication of code going on and will probably submit another pr with those changes.

**Pull request merged**